test/iterators.jl

```diff
@@ -242,6 +242,23 @@ end
     @test Base.IteratorEltype(typeof(dropwhile(<(4),Iterators.map(identity, 1:10)))) isa Base.EltypeUnknown
 end
 
+# findall
+# ----------------
+@testset "Iterators.findall" begin
+    let findall = Iterators.findall
+        f = findall(isnumeric, "abc257wf")
+        @test !(f isa AbstractArray) # it's lazy
+        @test collect(f) == [4,5,6]
+
+        f = findall(isodd, Dict(1 => 2, 2 => 4, 3 => 6))
+        @test_throws ArgumentError only(f)
```

Comment:
Could you comment on this testcase? Is it not the same (but more cryptic imho) as the test on the following line, `@test isempty(f)`?

Reply:
Changed to checking that `isnothing(iterate(x))`